From cc3d1c727d5b6444dfe14d10a1e18783612ba578 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:09:04 -0400
Subject: [acpi] scan: Set flags.match_driver in acpi_bus_scan_fixed()

Message-id: <20140710010904.22146.33345.stgit@gir.stowe>
Patchwork-id: 85642
O-Subject: [RHEL7.1 PATCH 103/105] ACPI / scan: Set flags.match_driver in acpi_bus_scan_fixed()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 883461679ad1ed6e09135adf8c02bb98d7962623

commit 883461679ad1ed6e09135adf8c02bb98d7962623
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Mon Nov 18 14:18:47 2013 +0100

    ACPI / scan: Set flags.match_driver in acpi_bus_scan_fixed()

    Before commit 6931007cc90b (ACPI / scan: Start matching drivers
    after trying scan handlers) the match_driver flag for all devices
    was set in acpi_add_single_object(), but now it is set by
    acpi_bus_device_attach() which is not called for the "fixed"
    devices added by acpi_bus_scan_fixed().  This means that
    flags.match_driver is never set for those devices now, so make
    acpi_bus_scan_fixed() set it before calling device_attach().

    Fixes: 6931007cc90b (ACPI / scan: Start matching drivers after trying scan h
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index d5c429e..52180c8 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -1976,6 +1976,7 @@ static int acpi_bus_scan_fixed(void)
   if (result)
    return result;
 
+  device->flags.match_driver = true;
   result = device_attach(&device->dev);
   if (result < 0)
    return result;
@@ -1992,6 +1993,7 @@ static int acpi_bus_scan_fixed(void)
   if (result)
    return result;
 
+  device->flags.match_driver = true;
   result = device_attach(&device->dev);
  }
 
-- 
1.7.1